include/gz/math/VolumetricGridLookupField.hh

```diff
@@ -143,6 +143,53 @@ namespace gz
           return interpolators;
         }
 
+        /// \brief Constructor
+        /// \param[in] _cloud The cloud of points to use to construct the grid.
+        /// \param[in] _indices A series of indices these points correspond to.
+        public: VolumetricGridLookupField(
+          const std::vector<Vector3<T>> &_cloud,
+          const std::vector<I> &_indices)
+        {
+          assert(_indices.size() == _cloud.size());
+          // NOTE: This part of the code assumes an exact grid of points.
+          // The grid may be distorted or the stride between different points
+          // may not be the same, but fundamentally the data is structured still
+          // forms a grid-like structure. It keeps track of the axis indices for
+          // each point in the grid.
+          for(auto pt : _cloud)
+          {
+            x_indices_by_lat.AddIndexIfNotFound(pt.X());
```

Comment:
@arjo129 nit^2, preexisting: lat, lon, depth imply a given set of coordinates that is not required by the API itself (it could easily be cartesian)